### PR TITLE
Upgrade react-syntax-highlighter to the latest version

### PR DESCRIPTION
**Pull request by stof**
Issue: #8208 #8038

## What I did

react-syntax-highlighter is updated to the latest version, which is using an uptodate version of regenerator-runtime rather than the old 0.11 one.

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? no

### CODE REVIEW

**Comment by stof** [comment]
needs https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38776 to be merged and released to fix typescript.

File: addons/storysource/src/StoryPanel.js
Comment (stof) [inline addons/storysource/src/StoryPanel.js]: should be checked again based on the comment in https://github.com/conorhastings/react-syntax-highlighter/pull/141#issuecomment-537125730

**Comment by ndelangen** [comment]
Very much appreciated if you could get everything fixed, and upgraded 👏👏👏

**Comment by stof** [comment]
@ndelangen as I commented, I need to wait for the merge of the bugfix for types, as that's the main blocker (typescript thinks the sub-module does not exist because types tell it so). This is what's breaking builds for Travis, CircleCI as well as deployments.

**Comment by stof** [comment]
@ndelangen any hint at what the DangerJS failed workflow means for my PR ?

**Comment by stof** [comment]
Looks like it is https://github.com/danger/danger-js/issues/918

**Comment by ndelangen** [comment]
@stof We use PR labels to generate changelogs. So it's important that before merging the correct labels are applied. this is what Danger warns us for. It blocks Merging until labels are applied correctly.

**Comment by shilman** [comment]
> $ node ../../scripts/prepare.js
src/syntaxhighlighter/syntaxhighlighter.tsx(6,17): error TS7016: Could not find a declaration file for module 'react-syntax-highlighter/dist/esm/languages/prism/jsx'. '/tmp/storybook/node_modules/react-syntax-highlighter/dist/esm/languages/prism/jsx.js' implicitly has an 'any' type.
  If the 'react-syntax-highlighter' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-syntax-highlighter`
src/syntaxhighlighter/syntaxhighlighter.tsx(7,18): error TS7016: Could not find a declaration file for module 'react-syntax-highlighter/dist/esm/languages/prism/bash'. '/tmp/storybook/node_modules/react-syntax-highlighter/dist/esm/languages/prism/bash.js' implicitly has an 'any' type.
  If the 'react-syntax-highlighter' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-syntax-highlighter`
src/syntaxhighlighter/syntaxhighlighter.tsx(8,17): error TS7016: Could not find a declaration file for module 'react-syntax-highlighter/dist/esm/languages/prism/css'. '/tmp/storybook/node_modules/react-syntax-highlighter/dist/esm/languages/prism/css.js' implicitly has an 'any' type.
  If the 'react-syntax-highlighter' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-syntax-highlighter`
src/syntaxhighlighter/syntaxhighlighter.tsx(9,18): error TS7016: Could not find a declaration file for module 'react-syntax-highlighter/dist/esm/languages/prism/markup'. '/tmp/storybook/node_modules/react-syntax-highlighter/dist/esm/languages/prism/markup.js' implicitly has an 'any' type.
  If the 'react-syntax-highlighter' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-syntax-highlighter`

**Comment by stof** [comment]
@ndelangen couldn't it report that in the github actions logs instead, which would work for fork PRs too ?

**Comment by stof** [comment]
@shilman I know about that, and I even commented about that in the PR previously.

I just pushed an update to that PR now that fixes for the types have been merged an released.

**Comment by stof** [comment]
It looks like the testsuite does not like ES6 imports in the dependencies. I could switch to importing the CommonJS files (with another update to the types, as these are not documented there), but that would be a bad news for tree-shaking in webpack builds.

@ndelangen any idea how to make Jest work here ?

**Comment by ndelangen** [comment]
Visual changes look very good!

**Comment by ndelangen** [comment]
> any idea how to make Jest work here ?

🤔

**Comment by ndelangen** [comment]
I think we can use this:
https://jestjs.io/docs/en/configuration#modulenamemapper-object-string-string

**Review by ndelangen — APPROVED** [review]

**Pull request merged**
